pdgraster/WMTSCapabilitiesGenerator.py

```diff
@@ -0,0 +1,247 @@
+        ET.SubElement(allowed_values, "ows:Value").text = "RESTful"
+
+    def _add_contents(self, root):
+        contents = ET.SubElement(root, "Contents")
+        layer = ET.SubElement(contents, "Layer")
+        ET.SubElement(layer, "ows:Title").text = "iwp_high"
+        ET.SubElement(layer, "ows:Identifier").text = "iwp_high"
+
+        wgs84_bbox = ET.SubElement(layer, "ows:WGS84BoundingBox")
+        ET.SubElement(wgs84_bbox, "ows:LowerCorner").text = f"{self.bounding_box['left']} {self.bounding_box['bottom']}"
+        ET.SubElement(wgs84_bbox, "ows:UpperCorner").text = f"{self.bounding_box['right']} {self.bounding_box['top']}"
+
+
+        style = ET.SubElement(layer, "Style", attrib={"isDefault": "true"})
+        ET.SubElement(style, "ows:Title").text = "Default Style"
+        ET.SubElement(style, "ows:Identifier").text = "default"
+
+        ET.SubElement(layer, "Format").text = "image/png"
+        tile_matrix_set_link = ET.SubElement(layer, "TileMatrixSetLink")
+        ET.SubElement(tile_matrix_set_link, "TileMatrixSet").text = "WGS1984Quad"
+
+        
+
+        resource_url = ET.SubElement(layer, "ResourceURL", attrib={
+            "format": self.tile_format,
+            "resourceType": "tile",
+            "template": self.resource_template
+        })
+
+        self._add_tile_matrix_set(contents)
+
+    def _add_tile_matrix_set(self, contents):
+        tile_matrix_set = ET.SubElement(contents, "TileMatrixSet", attrib={"xml:id": "WorldCRS84Quad"})
+        ET.SubElement(tile_matrix_set, "ows:Title").text = "CRS84 for the World"
+        ET.SubElement(tile_matrix_set, "ows:Identifier").text = "WGS1984Quad"
+
+        bounding_box = ET.SubElement(tile_matrix_set, "ows:BoundingBox", attrib={"crs": "http://www.opengis.net/def/crs/OGC/1.3/CRS84"})
+        ET.SubElement(bounding_box, "ows:LowerCorner").text = "-180 -90"
+        ET.SubElement(bounding_box, "ows:UpperCorner").text = "180 90"
+
+        ET.SubElement(tile_matrix_set, "ows:SupportedCRS").text = "http://www.opengis.net/def/crs/OGC/1.3/CRS84"
+        ET.SubElement(tile_matrix_set, "WellKnownScaleSet").text = "http://www.opengis.net/def/wkss/OGC/1.0/GoogleCRS84Quad"
+
+        for i in range(self.max_z_level + 1):  # Generate levels from 0 to max_z_level
+            tile_matrix = ET.SubElement(tile_matrix_set, "TileMatrix")
+            ET.SubElement(tile_matrix, "ows:Identifier").text = str(i)
+            ET.SubElement(tile_matrix, "ScaleDenominator").text = str(self.scale_denominator_base / (2 ** i))
+            ET.SubElement(tile_matrix, "TopLeftCorner").text = self.top_left_corner
+            ET.SubElement(tile_matrix, "TileWidth").text = str(self.tile_width)
+            ET.SubElement(tile_matrix, "TileHeight").text = str(self.tile_height)
+            ET.SubElement(tile_matrix, "MatrixWidth").text = str(2 ** (i+1))
+            ET.SubElement(tile_matrix, "MatrixHeight").text = str(2 ** i)
+
+# For testing purposes. Remove after
```

Comment:
Can this be moved into an e.g., integration test that asserts that the generated XML contains what is expected?

Reply:
That's a good point, I will move this part. Thank you!